src/net.h

```diff
@@ -1022,8 +1022,12 @@ class CConnman
 
     void SetAsmap(std::vector<bool> asmap) { addrman.m_asmap = std::move(asmap); }
 
-    /** Return true if the peer has been connected for long enough to do inactivity checks. */
-    bool RunInactivityChecks(const CNode& node) const;
+    /** Return true if we should disconnect the peer for failing an inactivity check. */
+    bool ShouldRunInactivityChecks(const CNode& node, const uint64_t const* pnow=nullptr) const
+    {
+        const uint64_t now = (pnow ? *pnow : GetSystemTimeInSeconds());
```

Comment:
I think you can use `std::optional<uint64_t>::value_or(uint64_t)` instead of passing pointers, no?

Reply:
Was there any reason to move this from net.cpp to net.h? (Coding guidelines say stuff should be in the .cpp unless inlining is critical)

I don't think `optnow.value_or(GetSystemTimeInseconds())` would be a good idea -- that would mean always invoking `GetSystemTimeInSeconds` and throwing away the result if you'd passed in a time.

Reply:
Ah good point on the arg to value_or always being evaluated. Though, keeping the ternary as is and changing the type to optional would work?

Reply:
> Was there any reason to move this from net.cpp to net.h? (Coding guidelines say stuff should be in the .cpp unless inlining is critical [EDIT: I assumed the coding guidelines were why you didn't make it inline in the first place])

Only because you and amiti seemed to want it inlined. I didn't realise that this guidance was in the developer notes, but I completely agree. net.h is included in just about everything so any lines of code in this file are compiled hundreds of times.

I'll move it back to the cpp file.

Reply:
@jnewbery Having it inline means the `?:` can be resolved at compile time for every use case, and I couldn't resist the temptation.

@MarcoFalke Yes, keeping the `?:` should work fine with optional instead of pointer.

Reply:
I don't have a strong opinion about any of this. My default is to implement things in the cpp file unless the function is in a tight loop where performance really matters and the function call overhead is meaningful. Your suggestion in the previous PR was to inline it, and amiti mentioned the same thing to me, so I made that change here.

I've now moved it back to the cpp and made the optional `now` arg a std::optional.

Reply:
Yeah, it wouldn't surprise me if it gets optimised during linking these days anyway. My bad habits don't need to be yours!

Reply:
I don't think we have link time optimization enabled yet, but hopefully in the future!